### PR TITLE
chore(geneators): add fake data and fake data generator

### PR DESCRIPTION
**Pull request by kaczor6418**
- [ ] fix data model (remove `users` collection + change id from `string` into `DocumentReferenve`)

**[IMPORTANT]: Run `npm install` after pulling this branch**

closes: #21

### CODE REVIEW

File: src/services/FirebaseService/OffersCollection/OffersCollectionService.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+): Promise<DocumentReference<OfferRecord>[]> {
+  const batch = writeBatch(firebaseDB);
+  const createdDocs: DocumentReference<OfferRecord>[] = [];
+  for (const offerData of offers) {
+    const newShop = doc(firebaseDB, Collection.OFFERS, crypto.randomUUID());
+    if (isNullOrUndefined(options)) {
+      batch.set(newShop, offerData);
+    } else {
+      batch.set(newShop, offerData, options);
+    }
```
Comment (HKGx) [inline src/services/FirebaseService/OffersCollection/OffersCollectionService.ts:26]: just having `batch.set(newShop, offerData, options)` won't work?

[3]

File: src/services/FirebaseService/ShopsCollection/ProductsService.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const newProduct = doc<Collection.SHOPS, SubCollection.PRODUCTS>(
+      parentShop,
+      SubCollection.PRODUCTS,
+      crypto.randomUUID()
+    );
+    if (isNullOrUndefined(options)) {
+      batch.set(newProduct, productData);
+    } else {
+      batch.set(newProduct, productData, options);
+    }
```
Comment (HKGx) [inline src/services/FirebaseService/ShopsCollection/ProductsService.ts:42]: just having `batch.set(newProduct, productData, options)` won't work?

[3]

File: src/services/FirebaseService/ShopsCollection/ShopsService.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+): Promise<DocumentReference<ShopRecord>[]> {
+  const batch = writeBatch(firebaseDB);
+  const createdDocs: DocumentReference<ShopRecord>[] = [];
+  for (const shopData of shops) {
+    const newShop = doc(firebaseDB, Collection.SHOPS, crypto.randomUUID());
+    if (isNullOrUndefined(options)) {
+      batch.set(newShop, shopData);
+    } else {
+      batch.set(newShop, shopData, options);
+    }
```
Comment (HKGx) [inline src/services/FirebaseService/ShopsCollection/ShopsService.ts:26]: just having `batch.set(newShop, shopData, options)` won't work?

[3]

**Review by HKGx — CHANGES_REQUESTED** [review]
Overall looks nice but I have some issues

I have grouped 3 issues with **[1]**, **[2]**, and **[3]** tags

File: mockDataGenerator/generateRandomData.utils.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+export async function getRandomPhotoUrl(width = 1920, height = 1080): Promise<string> {
+  const picsumResponse = await fetch(`https://picsum.photos/${width}/${height}`);
+
+  return picsumResponse.url;
+}
+
+export async function getRandomDescription(paragraphs = 1): Promise<string> {
+  const dinosResponse = await fetch(
+    `https://dinoipsum.com/api/?format=text&paragraphs=${paragraphs}`
+  );
```
[the remark this answers is not in the document]
Comment (kaczor6418) [reply mockDataGenerator/generateRandomData.utils.ts:99]: let's use this library
https://github.com/knicklabs/lorem-ipsum.js

**Pull request merged**